### DISCUSSION TITLE
WC v2 on React Native: ReferenceError: Can't find variable: TextEncoder

### DISCUSSION THREAD

**Discussion by khoanguyen-yang** [WalletConnect/walletconnect-monorepo#1589] [Developer Support]
Hi guys, I encountered this error when integrating WalletConnect V2 on React Native

I did add the polyfill `import '@walletconnect/react-native-compat'` but it does not work.

Really appreciate any help.

**Comment by CallbackDan** [comment]
Hi- did you find an answer to this? Thanks.

**Comment by hyrbis** [comment]
i have same problem and it shown when i place component with WC logic into router, but when i use functions of WC in App file all is fine :(
